pkg/apis/serving/v1alpha1/trainedmodel_webhook.go

```diff
@@ -27,16 +27,20 @@ import (
 
 // regular expressions for validation of isvc name
 const (
-	TmNameFmt                   string = "[a-z]([-a-z0-9]*[a-z0-9])?"
-	InvalidTmNameFormatError           = "the Trained Model \"%s\" is invalid: a Trained Model name must consist of lower case alphanumeric characters or '-', and must start with alphabetical character. (e.g. \"my-name\" or \"abc-123\", regex used for validation is '%s')"
-	InvalidTmMemoryModification        = "the Trained Model \"%s\" memory field is immutable. The memory was \"%s\" but it is updated to \"%s\""
+	TmNameFmt                    string = "[a-z]([-a-z0-9]*[a-z0-9])?"
+	StorageUriFmt                string = "(gs|s3)://(([a-z]([-a-z0-9]*[a-z0-9])?/))*([a-z]([-a-z0-9]*[a-z0-9]))"
```

Comment:
We probably should check against https://github.com/kubeflow/kfserving/blob/master/pkg/agent/storage/provider.go#L26 . So next time when we add a new provider we don't need to update both places.